domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/ShapePoint.java

```diff
@@ -0,0 +1,258 @@
+            // these fields could be provided as null values
+            //noinspection ConstantConditions to avoid lint
+            if (shapeId == null ||
+                    shapePtLat == null || (shapePtLat != null && !isValidLatitude()) ||
+                    shapePtLon == null || (shapePtLon != null && !isValidLongitude()) ||
+                    shapePtSequence == null || (shapePtSequence != null && shapePtSequence < 0) ||
+                    (shapeDistTraveled != null && shapeDistTraveled < 0)) {
+                if (shapeId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("shapes.txt", "shape_id",
+                            shapeId));
+                }
+                if (shapePtLat == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("shapes.txt", "shape_pt_lat",
+                            shapeId));
+                } else if (!isValidLatitude()) {
+                    noticeCollection.add(new FloatFieldValueOutOfRangeNotice("shapes.txt",
+                            "shape_pt_lat", shapeId, -90.0f, 90.0f, shapePtLat));
+                }
+                if (shapePtLon == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("shapes.txt", "shape_pt_lon",
+                            shapeId));
+                } else if (!isValidLongitude()) {
+                    noticeCollection.add(new FloatFieldValueOutOfRangeNotice("shapes.txt",
+                            "shape_pt_lon", shapeId, -180.0f, 180.0f, shapePtLon));
+                }
+                if (shapePtSequence == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("shapes.txt",
+                            "shape_pt_sequence",
+                            shapeId));
+                } else if (shapePtSequence < 0) {
+                    noticeCollection.add(new IntegerFieldValueOutOfRangeNotice("shapes.txt",
+                            "shape_pt_sequence", shapeId, 0, Integer.MAX_VALUE, shapePtSequence));
+                }
+                if (shapeDistTraveled != null && shapeDistTraveled < 0) {
+                    noticeCollection.add(new FloatFieldValueOutOfRangeNotice("shapes.txt",
+                            "shape_dist_traveled", shapeId, 0, Float.MAX_VALUE, shapeDistTraveled));
+                }
+                return new EntityBuildResult<>(noticeCollection);
+            } else {
+                return new EntityBuildResult<>(new ShapePoint(shapeId, shapePtLat, shapePtLon, shapePtSequence,
+                        shapeDistTraveled));
+            }
+        }
+
+        /**
+         * Utility method to determine the range validity of latitude
+         * @return true if the shapePtLat is valid, otherwise returns false
+         */
+        private boolean isValidLatitude() {
+            return shapePtLat > -90.0f && shapePtLat < 90.0f;
```

Comment:
Should be inclusive:

```suggestion
            return shapePtLat >= -90.0f && shapePtLat <= 90.0f;
```

Tests need to be updated too, if there are some.

Reply:
Actually these methods are not directly tested. However, the result they produce In `build` method is tested

Reply:
Do you think that these methods should be `public` or`protected` so that we can individually verify there implementation?

Reply:
I think these should be `public` methods centralized somewhere (a utility class?), as we know we'll need them for `stops.txt` too.

Reply:
As discussed, range verification is already performed at the parsing stage. Therefore, there is no need for these checks here. Removing this duplicate code.
Also, the utility class will be implemented in another PR.